# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../orsa/gcc41.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur orsa-0.7.0-orig/src/liborsa/orsa_body.h orsa-0.7.0/src/liborsa/orsa_body.h
--- orsa-0.7.0-orig/src/liborsa/orsa_body.h	2004-12-30 03:58:03.000000000 +0000
+++ orsa-0.7.0/src/liborsa/orsa_body.h	2006-04-03 21:31:50.423120000 +0000
@@ -183,7 +183,7 @@
     inline void SetVelocity(const double x, const double y, const double z) { Vector v(x,y,z); SetVelocity(v); }
     
     // b position - this position
-    inline Vector Body::distanceVector(const Body & b) const { return b.position()-position(); }
+    inline Vector distanceVector(const Body & b) const { return b.position()-position(); }
     inline double distance(const Body & b) const { return distanceVector(b).Length(); }
     
     // alias
diff -Nur orsa-0.7.0-orig/src/libxorsa/xorsa_analysis.h orsa-0.7.0/src/libxorsa/xorsa_analysis.h
--- orsa-0.7.0-orig/src/libxorsa/xorsa_analysis.h	2004-06-25 00:58:30.000000000 +0000
+++ orsa-0.7.0/src/libxorsa/xorsa_analysis.h	2006-04-03 21:44:48.123723250 +0000
@@ -129,7 +129,7 @@
   XOrsaPeaksListItem(QListView *parent, QString label1, QString label2 = QString::null, QString label3 = QString::null, QString label4 = QString::null, QString label5 = QString::null, QString label6 = QString::null, QString label7 = QString::null, QString label8 = QString::null);
   
  public:
-  int XOrsaPeaksListItem::compare(QListViewItem * i, int col, bool ascending) const;    
+  int compare(QListViewItem * i, int col, bool ascending) const;    
   
 };
 
diff -Nur orsa-0.7.0-orig/src/libxorsa/xorsa_import_astorb_objects.h orsa-0.7.0/src/libxorsa/xorsa_import_astorb_objects.h
--- orsa-0.7.0-orig/src/libxorsa/xorsa_import_astorb_objects.h	2005-01-05 03:04:17.000000000 +0000
+++ orsa-0.7.0/src/libxorsa/xorsa_import_astorb_objects.h	2006-04-03 21:39:40.844519500 +0000
@@ -600,7 +600,7 @@
   inline XOrsaAstorbObjectItem(QListView *parent, QString label1, QString label2 = QString::null, QString label3 = QString::null, QString label4 = QString::null, QString label5 = QString::null, QString label6 = QString::null, QString label7 = QString::null, QString label8 = QString::null) : QListViewItem(parent, label1, label2, label3, label4, label5, label6, label7, label8) { };
   
  public:
-  inline int XOrsaAstorbObjectItem::compare(QListViewItem *i, int col, bool ascending) const {
+  inline int compare(QListViewItem *i, int col, bool ascending) const {
 
     using std::atof;
         
diff -Nur orsa-0.7.0-orig/src/libxorsa/xorsa_object_selector.cc orsa-0.7.0/src/libxorsa/xorsa_object_selector.cc
--- orsa-0.7.0-orig/src/libxorsa/xorsa_object_selector.cc	2004-06-25 03:12:57.000000000 +0000
+++ orsa-0.7.0/src/libxorsa/xorsa_object_selector.cc	2006-04-03 21:50:05.967587250 +0000
@@ -40,7 +40,7 @@
   XOrsaObjectItem(QListView *parent, QString label1, QString label2 = QString::null, QString label3 = QString::null, QString label4 = QString::null, QString label5 = QString::null, QString label6 = QString::null, QString label7 = QString::null, QString label8 = QString::null);
   
  public:
-  int XOrsaObjectItem::compare(QListViewItem * i, int col, bool ascending) const;    
+  int compare(QListViewItem * i, int col, bool ascending) const;    
   
 };
 
diff -Nur orsa-0.7.0-orig/src/orsa/xorsa.h orsa-0.7.0/src/orsa/xorsa.h
--- orsa-0.7.0-orig/src/orsa/xorsa.h	2004-07-13 02:21:19.000000000 +0000
+++ orsa-0.7.0/src/orsa/xorsa.h	2006-04-03 21:56:22.087093250 +0000
@@ -62,7 +62,7 @@
   ObjectItem(QListView *parent, QString label1, QString label2 = QString::null, QString label3 = QString::null, QString label4 = QString::null, QString label5 = QString::null, QString label6 = QString::null, QString label7 = QString::null, QString label8 = QString::null);
   
  public:
-  int ObjectItem::compare(QListViewItem * i, int col, bool ascending) const;    
+  int compare(QListViewItem * i, int col, bool ascending) const;    
   
 };
 
